brewtarget: 2.3.1 -> 3.0.6

avnik opened the pull request:
Upgrade to 3.0.6
Also drop "insecure" qtwebkit

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

avnik committed:
brewtarget: 2.3.1 -> 3.0.6

Drop insecure qtwebkit

balsoft reviewed (changes requested):
Actually, please make sure that the package name in commit and PR title match the `pname` exactly, including the case

totoroot commented on pkgs/applications/misc/brewtarget/default.nix:
```diff
@@ -2,27 +2,29 @@
+, xercesc
 }:
 
 mkDerivation rec {
```
```suggestion
stdenv.mkDerivation rec {
```

totoroot commented on pkgs/applications/misc/brewtarget/default.nix:
```diff
@@ -2,27 +2,29 @@
+    sha256 = "sha256-qgwzRjRVEk8hM1CmkoWrJT285RQnk5zq16dfSijai1Q=";
   };
 
   nativeBuildInputs = [ cmake ];
```
```suggestion
  nativeBuildInputs = [ cmake wrapQtAppsHook ];
```

totoroot commented on pkgs/applications/misc/brewtarget/default.nix:
```diff
@@ -2,27 +2,29 @@
-, qtwebkit
 , qtsvg
+, xalanc
+, xercesc
```
```suggestion
, xercesc
, wrapQtAppsHook
```

totoroot commented on pkgs/applications/misc/brewtarget/default.nix:
```diff
@@ -2,27 +2,29 @@
 , mkDerivation
```
```suggestion
, stdenv
```

totoroot reviewed (changes requested):
@avnik Could you please also replace Qt's `mkDerivation` with `stdenv.mkDerivation` to fix
https://github.com/NixOS/nixpkgs/issues/180841
For that to work `wrapQtAppsHook` will be needed.

If you can, please also update it to 3.0.7 which has already been released in the meantime:
https://github.com/Brewtarget/brewtarget/releases/tag/v3.0.7

totoroot commented:
Also this PR would fix the issue with qtwebkit for brewtarget (for cross-reference):
https://github.com/NixOS/nixpkgs/issues/53079

totoroot commented on pkgs/applications/misc/brewtarget/default.nix:
```diff
@@ -2,27 +2,29 @@
 mkDerivation rec {
   pname = "brewtarget";
-  version = "2.3.1";
+  version = "3.0.6";
```
```suggestion
  version = "3.0.7";
```

totoroot commented on pkgs/applications/misc/brewtarget/default.nix:
```diff
@@ -2,27 +2,29 @@
     repo = pname;
     rev = "v${version}";
-    sha256 = "14xmm6f8xmvypagx4qdw8q9llzmyi9zzfhnzh4kbbflhjbcr7isz";
+    sha256 = "sha256-qgwzRjRVEk8hM1CmkoWrJT285RQnk5zq16dfSijai1Q=";
```
```suggestion
    hash = "sha256-UKdl716zTssSWRKeDCjqL/Yc1UCsvqwJTBbbUB8AVFs=";
```

avnik committed:
brewtarget: switch to use stdenv.mkDerivation

avnik commented:
@totoroot I fixed all suggestions (except bumping to 3.0.7 -- newer brewtarget depends on boost_stacktrace_backtrace which not yet packaged), also rebased PR on newer master

totoroot commented:
Result of `nixpkgs-review pr 217648` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

totoroot commented:
@avnik Sorry for the late response. The last I checked, it did not build on darwin, which can be seen in ofborg's tests as well.
I have to check again later this week, what has to be done exactly to make it build.
Meanwhile [brewtarget 3.0.9 got released](https://github.com/Brewtarget/brewtarget/releases/tag/v3.0.9)...would you mind updating the PR?

doronbehar commented on pkgs/top-level/all-packages.nix:
```diff
@@ -2958,7 +2958,7 @@ with pkgs;
 
   brakeman = callPackage ../development/tools/analysis/brakeman { };
 
-  brewtarget = libsForQt5.callPackage ../applications/misc/brewtarget { } ;
```
Why?
